### PR TITLE
Implicitly force import manually added user

### PR DESCRIPTION
**Pull request by nfx**
Add `force` parameter to `databricks_user`, so that SCIM sync corner cases might be simplified

image

### CODE REVIEW

**Comment by alexott** [comment]
Should we support the same for groups? It would help when switching from SCIM connector…

File: scim/resource_user.go [hunk 1]
```diff
@@ -70,3 +77,23 @@ func ResourceUser() *schema.Resource {
...
 	}.ToResource()
 }
+
+func createForceOverridesManuallyAddedUser(err error, d *schema.ResourceData, usersAPI UsersAPI, u User) error {
+	forceCreate := d.Get("force").(bool)
+	if !forceCreate {
+		return err
+	}
+	// corner-case for overriding manually provisioned users
+	userName := strings.ReplaceAll(u.UserName, "'", "")
```
Comment (alexott) [inline scim/resource_user.go:87]: I think that it doesn’t track sanitizing - should we put where it’s used?

**Review by alexott — APPROVED** [review]
.look good

File: scim/resource_user.go [hunk 2]
```diff
@@ -40,9 +45,25 @@ func ResourceUser() *schema.Resource {
...
+			usersAPI := NewUsersAPI(ctx, c)
+			user, err := usersAPI.Create(u)
 			if err != nil {
-				return err
+				forceCreate := d.Get("force").(bool)
+				if !forceCreate {
+					return err
+				}
+				// corner-case for overriding manually provisioned users
+				force := fmt.Sprintf("User with username %s already exists.", u.UserName)
```
[the remark this answers is not in the document]
Comment (nfx) [reply scim/resource_user.go]: Probably they won't. I considered relying on 409 status code, but it's returned for other things as well

File: scim/resource_user.go [hunk 3]
```diff
@@ -40,9 +45,25 @@ func ResourceUser() *schema.Resource {
...
+				if err.Error() != force {
+					return err
+				}
+				userList, err := usersAPI.Filter(fmt.Sprintf("userName eq '%s'", u.UserName))
+				if err != nil {
+					return err
+				}
+				user = userList[0]
```
[the remark this answers is not in the document]
Comment (nfx) [reply scim/resource_user.go]: If user is reported as existing? Illogical 🤷🏻‍♂️

**Pull request merged**